scripts/docker/docker-compose.yaml

```diff
@@ -0,0 +1,33 @@
+services:
+  metal:
+    build:
+      dockerfile: ../../dockerfile/tt-metal-x86.Dockerfile
+      context: .
+      args:
+        UID: ${HOST_UID}
+        GID: ${HOST_GID}
+    image: tenstorrent:tt-metal
+    container_name: tt-metal
+    environment:
+      - USER=${USER}
+    user: "${HOST_UID}:${HOST_GID}"
+    working_dir: /home/ubuntu
+    # Allow attaching to the container.
+    stdin_open: true
+
+    shm_size: "4gb"
+    devices:
+      - /dev/tenstorrent:/dev/tenstorrent
+    volumes:
+      - /dev/hugepages-1G:/dev/hugepages-1G
+      - /etc/group:/etc/group:ro
+      - /etc/passwd:/etc/passwd:ro
+      - /etc/shadow:/etc/shadow:ro
+      # Allow for persist file storage
+      - /home/${USER}/workspace:/home/ubuntu:rw
```

Comment:
when user enters docker, their username changes to ubuntu?

Reply:
yes, but they should have the same permissions as the user itself